docs/HazardAnalysis/HazardAnalysis.tex

```diff
@@ -113,6 +113,54 @@ \section{Safety and Security Requirements}
 \wss{Newly discovered requirements.  These should also be added to the SRS.  (A
 rationale design process how and why to fake it.)}
 
+\begin{enumerate}[label=SR\arabic*.]
+
+     \item The system shall allow users to reset their password without logging in and without knowledge of their current password.
+    \item[] \emph{Rationale:} Users cannot be relied on to remember their login credentials correctly. There must be a method of allowing the user to access their account despite this.
+    \item[] \emph{Associated Hazards:} HA1, HA2
+
+    \item The system shall limit the number of login attempts within a period of time before login is disabled for a cooldown period.
+    \item[] \emph{Rationale:} Malicious actors may gain access to an account by a brute-force attack. This requirement significantly reduces this possibility.
+    \item[] \emph{Associated Hazards:} HA2
+
+    \item The system shall use natural language processing to reduce the error generated by OCR reading of the receipt.
+    \item[] \emph{Rationale:} OCR can sometimes return an inaccurate reading of its input. Natural language processing can reduce this by interpreting the reading using its model rather than its literal reading.
+    \item[] \emph{Associated Hazards:} HB1
+
+    \item The system shall prompt the user to retake a photo if the receipt reading is unclear.
+    \item[] \emph{Rationale:} OCR will return incorrect data to the rest of the system if the photo is unclear. Thus requesting the user to retake the photo can mitigate this issue.
+    \item[] \emph{Associated Hazards:} HB1
+
+    \item The system shall allow the user to manually input grocery items.
+    \item[] \emph{Rationale:} Should the OCR and language model not read or recognize an item properly, the user can eliminate this error state by manually reviewing and inputting their items.
+    \item[] \emph{Associated Hazards:} HB1
+
+    \item The system shall prompt the user to grant system access to the device camera and the photo gallery.
+    \item[] \emph{Rationale:} If the application does not have camera access, it cannot take a photo. The prompt will notify the user that they must grant access before the application can take or import a photo.
+    \item[] \emph{Associated Hazards:} HB2
+
+    \item The system shall perform data validation on both front-end when the user inputs and back-end.
```

Comment:
Wording sounds awkward to me. Maybe "The system shall perform data validation on both front-end and back-end for user inputs"?